app/keepers/keepers.go

```diff
@@ -577,6 +579,16 @@ func NewAppKeeper(
 		govModAddress,
 	)
 
+	// Register the precompiled contracts
+	precompile.RegistPrecompiledContract(
+		appKeepers.AccountKeeper,
+		appKeepers.BankKeeper,
+		stakingkeeper.NewMsgServerImpl(appKeepers.StakingKeeper.Keeper),
+		distrkeeper.NewMsgServerImpl(appKeepers.DistrKeeper),
+		wasmkeeper.NewMsgServerImpl(&appKeepers.WasmKeeper),
```

Comment:
Since you are receiving appKeepers.WasmKeeper anyway at the bottom, why did you choose to add this line directly instead of executing wasmkeeper.NewMsgServerImpl(&appKeepers.WasmKeeper) inside the precompile.RegistPrecompiledContract function after receiving appKeepers.WasmKeeper? Is there a possibility that something other than &appKeepers.WasmKeeper might be passed as a parameter to wasmkeeper.NewMsgServerImpl?

Reply:
stakingkeeper and distributeKeeper only need msgServer, so I did that for consistency. 
I think it would be okay to receive them all as keeper and process them as server and keeper in RegisterPrecomplie.